framework/src/modules/chain/block_processor_v2.js

```diff
@@ -230,7 +240,33 @@ class BlockProcessorV2 extends BaseBlockProcessor {
 				this.blocksModule.verify(block, stateStore, { skipExistingCheck }),
 			({ block, stateStore }) => this.blocksModule.apply(block, stateStore),
 			({ block, tx }) => this.dposModule.apply(block, { tx }),
-			({ block, tx }) => this.bftModule.addNewBlock(block, tx),
+			async ({ block, tx }) => {
+				// We only need activeMinHeight value of the delegate who is forging the block.
+				// Since the block is always the latest,
+				// fetching only the latest active delegate list would be enough.
+				const numberOfRounds = 1;
+				const delegateMinHeightActiveList = await this.dposModule.getActiveDelegateHeights(
+					numberOfRounds,
+					{ tx },
+				);
+
+				const delegate = delegateMinHeightActiveList.find(
```

Comment:
I think you confused yourself by calling this ActiveList, since this is a map.
It would be simpler to do
```
{
  'delegatePublicKey': [active heights]
}
```
and you can probably do
`const [delegateMinHeightActive ] = delegateMinHeightActiveMap[block.generatorPublicKey]`